sdk/search/search/samples/typescript/src/readonlyQuery.ts

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * Performs a query over a public dataset
+ */
+
+import { SearchIndexClient, SearchApiKeyCredential, odata, GeographyPoint } from "@azure/search";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+interface Hotel {
+  HotelId: string;
+  HotelName: string;
+  Description: string;
+  Description_fr: string;
+  Category: string;
+  Tags: string[];
+  ParkingIncluded: boolean;
+  LastRenovationDate: Date;
+  Rating: number;
+  Address: {
+    StreetAddress: string;
+    City: string;
+    StateProvince: string;
+    PostalCode: string;
+    Country: string;
+  };
+  Location: GeographyPoint;
+  Rooms: Array<{
+    Description: string;
+    Description_fr: string;
+    Type: string;
+    BaseRate: number;
+    BedOptions: string;
+    SleepsCount: number;
+    SmokingAllowed: boolean;
+    Tags: string[];
+  }>;
+}
+
+export async function main() {
+  console.log(`Running readonly query sample`);
+
+  // Variables provided by https://docs.microsoft.com/en-us/samples/azure-samples/azure-search-sample-data/azure-search-sample-data/
+  const endpoint = "https://azs-playground.search.windows.net";
+  const apiKey = "EA4510A6219E14888741FCFC19BFBB82";
+  const indexName = "hotels";
+
+  const credential = new SearchApiKeyCredential(apiKey);
+  const client = new SearchIndexClient<Hotel>(endpoint, indexName, credential);
```

Comment:
I hadn't thought of parameterizing the client instead of the search/suggest methods themselves. Is that an option? It would provide more flexibility. I can imagine customers using different types to access different fields of a sparse index that represents different types of entities.

Reply:
I can look into if this is possible to do with the Pick for selected fields (that's why I did it this way for preview1), but if a developer wanted to do this today they could do something like:

```ts

interface Hotel {
  Type: "hotel";
  Id: string;
  Description: string;
  RoomCount: number;
}

interface ConferenceCenter {
  Type: "ConferenceCenter";
  Id: string;
  Description: string;
  AttendeeCapacity: number;
}

type IndexType = Hotel | ConferenceCenter;

const client = new SearchIndexClient<IndexType>(...)
```

Then when they get a result they can check `result.Type` to discriminate types. They could also use a [user defined type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards).

Reply:
Very cool!